diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8147d01..21b5a97 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,16 +7,13 @@
 cmake_minimum_required(VERSION 2.8)
 project(Uri)
 
-find_package(Threads REQUIRED)
-
 if(CPP-NETLIB_BUILD_SHARED_LIBS OR BUILD_SHARED_LIBS)
   set(Boost_USE_STATIC_LIBS OFF)
 else()
   set(Boost_USE_STATIC_LIBS ON)
 endif()
 
-set(Boost_USE_MULTITHREADED ON)
-find_package(Boost 1.53 REQUIRED system filesystem)
+set(Boost_USE_MULTITHREADED OFF)
 
 include_directories(${Uri_SOURCE_DIR}/src ${Boost_INCLUDE_DIRS})
 
@@ -50,13 +47,6 @@ endif(WIN32)
 
 add_subdirectory(src)
 
-# Testing
-if (CMAKE_PROJECT_NAME STREQUAL "Uri" OR CPP-NETLIB_BUILD_TESTS)
-  enable_testing()
-  find_package(GTest REQUIRED)
-  add_subdirectory(test)
-endif()
-
 # Documentation
 find_package(Doxygen)
 if (DOXYGEN_FOUND)
